pkg/tsdb/cloud-monitoring/httpclient.go

```diff
@@ -59,7 +59,13 @@ func getMiddleware(model *datasourceInfo, routePath string) (httpclient.Middlewa
 	return tokenprovider.AuthMiddleware(provider), nil
 }
 
-func newHTTPClient(model *datasourceInfo, opts httpclient.Options, clientProvider infrahttp.Provider, route string) (*http.Client, error) {
+type Provider interface {
```

Comment:
Can you use the SDK http provider for this as Azure does? https://github.com/grafana/grafana/blob/main/pkg/tsdb/azuremonitor/azuremonitor.go/#L28
If not, we should move this interface definition to the SDK rather than copying it here.